packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -138,41 +138,27 @@ export const AlephiumConnectProvider: React.FC<{
     return undefined
   }, [network, addressGroup, keyType])
 
-  const connectionStatusRef = useRef<ConnectionStatus>('connecting')
+  const [connectionStatus, setConnectionStatus] = useState<ConnectionStatus>('connecting')
```

Comment:
`useRef` does not rerender when connection status changes, so I replaced it with `useState` here. This means that there will be a rerender when the connection status changes.

If the initial status is `disconnected`, in the `useWallet` hook, the account will be undefined, so I set the initial status to `connecting`.

Reply:
We could initialize it like this:
```
  const [connectionStatus, setConnectionStatus] = useState<ConnectionStatus>(
    lastConnectedAccount ? 'connecting' : 'disconnected'
  )
```